ThirdPartyAdapters/snap/snap/src/main/java/com/google/ads/mediation/snap/SnapBannerAd.java

```
@@ -0,0 +1,80 @@
+package com.google.ads.mediation.snap;
+
+import android.os.Bundle;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.android.gms.ads.AdError;
+import com.google.android.gms.ads.mediation.MediationAdLoadCallback;
+import com.google.android.gms.ads.mediation.MediationBannerAd;
+import com.google.android.gms.ads.mediation.MediationBannerAdCallback;
+import com.google.android.gms.ads.mediation.MediationBannerAdConfiguration;
+import com.snap.adkit.external.BannerView;
+import com.snap.adkit.external.LoadAdConfig;
+import com.snap.adkit.external.LoadAdConfigBuilder;
+import com.snap.adkit.external.SnapAdClicked;
+import com.snap.adkit.external.SnapAdEventListener;
+import com.snap.adkit.external.SnapAdKitEvent;
+import com.snap.adkit.external.SnapAdLoadFailed;
+import com.snap.adkit.external.SnapAdLoadSucceeded;
+import com.snap.adkit.external.SnapAdSize;
+
+public class SnapBannerAd implements MediationBannerAd {
+    private static final String SLOT_ID_KEY = "adSlotId";
+
+    private BannerView mBannerView;
+
+    private MediationBannerAdConfiguration adConfiguration;
+    private MediationAdLoadCallback<MediationBannerAd, MediationBannerAdCallback> callback;
+    private MediationBannerAdCallback mBannerAdCallback;
+    private String mSlotId;
+
+    public SnapBannerAd(MediationBannerAdConfiguration adConfiguration,
+                               MediationAdLoadCallback<MediationBannerAd, MediationBannerAdCallback> callback) {
+        this.adConfiguration = adConfiguration;
+        this.callback = callback;
+    }
+
+    public void loadAd() {
+        mBannerView = new BannerView(adConfiguration.getContext());
+        mBannerView.setAdSize(SnapAdSize.BANNER);
```

Does snap support other Banner ad sizes?

---

no

---

We should check is the ad size from `adConfiguration` valid or not, using the `getAdSize` method.

See: https://developers.google.com/android/reference/com/google/android/gms/ads/mediation/MediationBannerAdConfiguration

---

+1
If the publisher requests for a different ad size (aka 300x250), we should return an error saying they are requesting for an unsupported ad size.

---

@changlileo Gentle bump on this.